### DISCUSSION TITLE
Ubuntu version?

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#4529] [Q&A] [answered]
The following is from a terminal within the IDE based on [this docker image](https://hub.docker.com/r/codercom/code-server):

    # lsb_release -a
    No LSB modules are available.
    Distributor ID: Debian
    Description:    Debian GNU/Linux 11 (bullseye)
    Release:        11
    Codename:       bullseye

Is there a version of the image based on Ubuntu instead?

I did try deploying with a Dockerfile that is based on an Ubuntu image, and installing code-server via [the script](https://coder.com/docs/code-server/latest/install#installsh), but ran into problems ([see here](https://github.com/caprover/caprover/issues/1266) for a full explanation).

**Comment by jsjoeio** [comment, marked as answer]
Not currently, but you can cast your vote on [this issue](https://github.com/cdr/code-server/issues/2849)!

**Reply by drmrbrewer** [reply]
OK, will do.  Do you have any thoughts on the problems I am having trying to install code-server "manually" within a Dockerfile based off an Ubuntu image... that issue is linked in my post.  I'm not even sure that I'll end up with a code-server with Ubuntu internally, even [using the script](https://coder.com/docs/code-server/latest/install#installsh) within an Ubuntu-based container, or whether it'll still be Debian.  I do notice that the folks over at linuxserver have an [ubuntu-based image](https://hub.docker.com/r/linuxserver/code-server/tags) `linuxserver/code-server:focal`, which may suit my purposes, but a slight niggle there is the inability to log in as `root`.

**Reply by jsjoeio** [reply]
Goooood question. I haven't tried using the install script inside Docker 🤔 

> reated symlink /etc/systemd/system/default.target.wants/code-server@root.service → /lib/systemd/system/code-server@.service.
> System has not been booted with systemd as init system (PID 1). Can't operate.

Hmm...did you try [this already](https://github.com/cdr/code-server/discussions/2714#discussioncomment-360160)?

@im-coder-lg or @code-asher might know!

**Reply by drmrbrewer** [reply]
I'm not sure what WSL is and how it might be relevant in my context 😕

**Reply by jsjoeio** [reply]
> I'm not sure what WSL is

[WSL - Ubuntu Wiki](https://wiki.ubuntu.com/WSL).

> how it might be relevant in my context

Unless we're talking about different things, they ran into the exact same "System has not been booted with systemd as init system" error. I'd advice you to take a look at the discussions and the links in the answer to the discussion!

**Reply by drmrbrewer** [reply]
> I'd advice you to take a look at the discussions and the links in the answer to the discussion!

I have, several times now.  The error is the same but the context is very different (installation in docker container vs installation in WSL2).  Slightly over my head.  Some discussion on stackoverflow (e.g. [here](https://stackoverflow.com/a/62489079/4070848) and [here](https://stackoverflow.com/a/51985145/4070848)) suggests that running `systemctl` within a docker container is a no-no, but I don't know enough about this to pick my way around it.  😢  All I know is that it's possible to run code-server out of a docker container without running `systemctl` or maybe running `systemctl` in a different way (because you guys have a docker image which works)... just want to know how this might be done when substituting debian for ubuntu (which are close enough that it should be possible with a tweak or two).

**Reply by jsjoeio** [reply]
>  but I don't know enough about this to pick my way around it. 😢 

I wish I knew more about this to help, but sadly, this is out of my wheelhouse 😢 

> just want to know how this might be done when substituting debian for ubuntu

Hmm...I feel like it should _just work_ but I'm not fully sure. @code-asher will be back in a week and might be able to help!

**Reply by lg-nineoneninethreenine** [reply]
Well, I feel like I saw someone list out a lot of commands, used Debian Slim(is that what it's called?) and configured it to install+expose code-server. Will find that and put it here.

PS: You could use Ubuntu instead of Debian and test it.

Edit: that someone posted the code as a discussion in this repo. Appeared on my notification feed.

**Reply by lg-nineoneninethreenine** [reply]
`systemctl` can never enable for code-server(or any apps in fact), I have tried it a lot of times but it failed. I think I read an article that Microsoft is going to enable `systemctl` on WSL 3, so if it's true, we need WSL 3, mostly on Windows 11.

**Reply by lg-nineoneninethreenine** [reply]
> System has not been booted with systemd as init system (PID 1). Can't operate.

I've seen this error a lot of times. Basically once I tried setting up SSH for my WSL system(preferably Ubuntu) but this popped up. I tried Genie but to no avail. Maybe the best fix is to get Ubuntu Server as a VM on your computer.

**Reply by lg-nineoneninethreenine** [reply]
Looks like getting Ubuntu Server as a VM might do the job! I have a Ubuntu Server LTS installed as a VM on my PC, it works. Sure, installation was kind of a pain but unless you are getting Ubuntu Server on an actual server, it's kinda easy. Just keep hitting Done, use only the keyboard, and go through the setup.

**Reply by drmrbrewer** [reply]
I'm not installing anything on my PC, though.  This is specifically a dockerised arrangement deployed on a cloud server running Ubuntu.  It must be possible to run code-server inside a docker container, accessible outside, because there is already a docker image (Debian) for it.  Likewise, the folks at linuxserver have an Ubuntu image for it, so perhaps I should use that.  My preference however would be to run a docker container that is based off the official Ubuntu docker image and then install the stuff I need into it, including code-server.

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Wonder how to add this!

**Comment by code-asher** [comment]
I am not sure if it is possible to make systemctl work in Docker.  The way we do it in our Docker image is to run code-server directly.

Here is a very minimal example for an Ubuntu-based Dockerfile. 

```
FROM ubuntu

RUN apt-get update \
 && apt-get install -y \
    curl

RUN curl -fsSL https://code-server.dev/install.sh | sh

ENTRYPOINT ["code-server", "--bind-addr", "0.0.0.0:8080"]
```

**Reply by drmrbrewer** [reply]
Wow, so simple when you know how!  This works perfectly, thanks!

**Comment by drmrbrewer** [comment]
Update almost a year later... looks like an Ubuntu-based image is available as from [version `4.8.0`](https://github.com/coder/code-server/releases/tag/v4.8.0)... nice!

**Reply by jsjoeio** [reply]
🤣 better late than never! All thanks to @huweiATgithub
